internal/olm/client/client.go

```diff
@@ -226,47 +258,60 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 
 	err := wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
 	if err != nil && errors.Is(err, context.DeadlineExceeded) {
-		if depCheckErr := c.printDeploymentErrors(ctx, key, csv); depCheckErr != nil {
-			return fmt.Errorf("error printing operator resource errors: %v %v", err, depCheckErr)
+		depCheckErr := c.checkDeploymentErrors(ctx, key, csv)
+		if depCheckErr != nil {
+			return depCheckErr
 		}
 	}
 	return err
 }
 
-// TODO(btenneti) Refactor function to collect errors into customized error and return.
-// printDeploymentErrors function loops through deployment specs of a given CSV, and prints reason
+// checkDeploymentErrors function loops through deployment specs of a given CSV, and prints reason
 // in case of failures, based on deployment condition.
-func (c Client) printDeploymentErrors(ctx context.Context, key types.NamespacedName, csv olmapiv1alpha1.ClusterServiceVersion) error {
+func (c Client) checkDeploymentErrors(ctx context.Context, key types.NamespacedName, csv olmapiv1alpha1.ClusterServiceVersion) error {
+	depErr := deploymentErrors{}
 	if key.Namespace == "" {
 		return fmt.Errorf("no namespace provided to get deployment failures")
 	}
 	dep := &appsv1.Deployment{}
 	for _, ds := range csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+		podErr := podErrors{}
```

Comment:
```suggestion
```

Reply:
This is needed, as podErr struct needs to be empty for each new Deployment spec in the loop.

Reply:
Not if you instantiate `depErr` as I’ve suggested it below.